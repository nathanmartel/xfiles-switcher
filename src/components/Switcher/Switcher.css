/* CSS heavily sourced from Ryan Mehta â€” gitHub: @mehtaphysical */
.Switcher {
  input {
    display: none;
  }

  label {
    display: block;
    width: 2rem;
    height: 1rem;
    border-radius: 1rem;
    border: 1px solid black;

    background-color: #FFFFFF;
  }

  label::before {
    content: '';
    display: block;
    width: 1rem;
    height: 1rem;
    border-radius: 100%;

    background-color: black;

    transition: 0.2s;
  }

  input:checked + label {
    background-color: green;

    &::before {
      margin-left: 1rem;
    }
  }
}
