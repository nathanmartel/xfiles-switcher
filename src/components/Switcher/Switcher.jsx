import React from 'react';
import PropTypes from 'prop-types';

const Switcher = ({ value, onSwitcherChange }) => (
  <section>
    <input id="switcher" type="checkbox" checked={value} onChange={onSwitcherChange} />
  </section>
);

Switcher.propTypes = {
  value: PropTypes.bool.isRequired,
  onSwitcherChange: PropTypes.func.isRequired
};

export default Switcher;
