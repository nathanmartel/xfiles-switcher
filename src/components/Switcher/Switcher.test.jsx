import React from 'react';
import { shallow } from 'enzyme';
import Switcher from './Switcher';

describe('Switcher component', () => {
  it('renders Switcher', () => {
    const wrapper = shallow(<Switcher value={true} onSwitcherChange={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
