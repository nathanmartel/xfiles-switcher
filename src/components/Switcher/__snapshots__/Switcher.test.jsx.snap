// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switcher component renders Switcher 1`] = `
<section
  className="Switcher"
>
  Main 
  <input
    checked={true}
    id="switcher"
    onChange={[Function]}
    type="checkbox"
  />
   Monster
  <label
    htmlFor="switcher"
  />
</section>
`;
